packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -467,28 +473,51 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		blob: ArrayBufferLike,
 	): Promise<ICreateBlobResponse | void> {
 		return runWithRetry(
-			async () => this.getStorage().createBlob(blob),
+			async () => {
+				try {
+					return await this.getStorage().createBlob(blob);
+				} catch (error) {
+					const entry = this.pendingBlobs.get(localId);
+					assert(
+						!!entry,
+						0x387 /* Must have pending blob entry for blob which failed to upload */,
+					);
+					if (entry.opsent && !canRetryOnError(error)) {
+						if (isFluidError(error)) {
+							error.addTelemetryProperties({ forceRetry: true });
```

Comment:
addTelemetryProperties can't overwrite them. I thought about a couple of ways to tackle it and this one avoids adding a lot of unnecessary changes in the error types.

Reply:
the thing the wrap error code below will do the right thing in all cases. i think we can just remove the if, and only do that